File: app/src/main/java/com/antyzero/smoksmog/ui/screen/order/dialog/StationViewHolder.java
package com.antyzero.smoksmog.ui.screen.order.dialog;

import android.support.v7.widget.RecyclerView;
import android.widget.TextView;

import pl.malopolska.smoksmog.model.Station;

public class StationViewHolder extends RecyclerView.ViewHolder {

    private final TextView textView;
    private final StationClickListener stationClickListener;

    public StationViewHolder( TextView textView, StationClickListener stationClickListener ) {
        super( textView );
        this.textView = textView;
        this.stationClickListener = stationClickListener;
    }

    public void bind( Station station ) {
        textView.setText( station.getName() );
        itemView.setOnClickListener( v -> stationClickListener.onClick( station.getId() ) );
    }

    public interface StationClickListener {

        void onClick( long stationId );
    }
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/order/dialog/StationViewHolder.kt
package com.antyzero.smoksmog.ui.screen.order.dialog

import android.support.v7.widget.RecyclerView
import android.widget.TextView
import pl.malopolska.smoksmog.model.Station

class StationViewHolder(private val textView: TextView, private val stationClickListener: StationViewHolder.StationClickListener) : RecyclerView.ViewHolder(textView) {

    fun bind(station: Station) {
        textView.text = station.name
        itemView.setOnClickListener { stationClickListener.onClick(station.id) }
    }

    interface StationClickListener {

        fun onClick(stationId: Long)
    }
}
